app/controllers/application_controller.rb

```diff
@@ -7,7 +7,23 @@ class ApplicationController < ActionController::Base
   before_filter :authorize!
   before_filter :fedora_setup
 
-  rescue_from ActiveFedora::ObjectNotFoundError, with: -> { render text: 'Object Not Found', status: :not_found }
+  rescue_from ActionController::BadRequest do
+    msg = "Bad request: #{params}"
+    logger.error msg
+    render text: msg, status: :bad_request
+  end
+
+  rescue_from ActiveFedora::ObjectNotFoundError do
+    msg = "Object Not Found: #{params}"
+    logger.error msg
+    render text: msg, status: :not_found
+  end
+
+  rescue_from Rubydora::FedoraInvalidRequest do
+    msg = "Fedora Rejected Object Request: #{params}"
+    logger.error msg
+    render text: msg, status: :not_found
```

Comment:
On my laptop a request for `find_druid('junk')` will hit the `Rubydora` exception and, yes, I agree it should be a 400.  On travis, it hits an `ActiveFedora::ObjectNotFoundError` and that should be a 404.  So, the specs are somehow platform dependent and I chose to coerce them into the same response code to avoid the platform specific inconsistency.